### DISCUSSION TITLE
Welcome to website Discussions!

### DISCUSSION THREAD

**Discussion by dimpase** [sagemath/website#226] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by subho951m** [comment]
@haraldschilly Sir you wrote the render.py of sagemath website? Can you please tell me which jinja2 contains pass_context function?
Everytime I am typing `make` command it's render.py which is making issue to locally view the sage math website.
![Screenshot (1)](https://user-images.githubusercontent.com/88569334/144274471-fb0c9efc-c954-48b5-823d-e00eec0421ca.png)
Can you please help me set up the website locally on my desktop so that I can continue development.
Please help.

**Comment by subho951m** [comment]
@haraldschilly I fresh cloned the repository but same thing is happening again and again.

**Comment by haraldschilly** [comment]
@subho951m it's not a good practice to reply to a thread with a different topic. Yes, I came up with that render.py file, but that's a long time ago.  despite that, I just searched for this and found it here: https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.pass_context

So, well, my first question is, what version of jinja2 are you using?

The travis-ci setup is here: https://github.com/sagemath/website/blob/master/.travis.yml and at least last time I checked, it was still working fine.

**Comment by haraldschilly** [comment]
Ah, and I have no recollection, but the line is part of [that commit](https://github.com/sagemath/website/commit/338c4412185912145b9c7a05ba3df60d4f56533b) ... interestingly, I must have seen deprecation warnings and acted preemptive to avoid this from breaking. So, my guess is, your jinja2 version is too old.

**Comment by subho951m** [comment]
@haraldschilly Sir I checked @j2.contextfilter & @j2.evalcontextfilter , they both are working fine. While @j2.pass_context & @j2.pass_eval_context is creating issues. Now with @j2.contextfilter the website can be viewed locally. Thanks for the help sir.

**Reply by subho951m** [reply]
The newer version of Jinja2 and all other packages works with only @j2.contextfilter & @j2.evalcontextfilter.

**Comment by subho951m** [comment]
@haraldschilly Sir `make` is working fine. I have another issue. Where should I ask any questions or issues related to website ?
I am a complete beginner to open source. Please guide me.
Sir Dima Pasechnik guided me to this thread.
For now please let me post the question here.
`make` ,`make open` is working fine with python 3 but in `make server` it requires python2. So I have to manually switch between python 3 to python 2 on Ubuntu 20.04(LTS). Is there any other alternative?
![Screenshot (5)](https://user-images.githubusercontent.com/88569334/144374048-0a138ce6-2aff-469a-bbbd-21ad22840a24.png)
While searching for this I came to know that `SimpleHTTPServer` has been merged to `http.server` in python3.
Is there any way available that I don't have to switch manually?
![Screenshot (4)](https://user-images.githubusercontent.com/88569334/144374382-dea972a0-a4d4-41f7-b410-5067d1722094.png)
